src/Datastore/DatastoreClient.php

```diff
@@ -137,7 +137,8 @@ public function __construct(array $config = [])
             'namespaceId' => null,
             'returnInt64AsObject' => false,
             'scopes' => [self::FULL_CONTROL_SCOPE],
-            'projectIdRequired' => true
+            'projectIdRequired' => true,
+            'hasEmulator' => (bool) getenv('DATASTORE_EMULATOR_HOST')
```

Comment:
Rather than setting `projectIdRequired` to `false`, I added a new setting because the emulator will still require a project ID. This setting will simply provide a dummy project ID if the user doesn't provide one.